src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -220,14 +237,25 @@ public override Task AnalyzeProjectAsync(Project project, bool semanticsChanged,
 
             private Task UpdateReferencesAync(Project project, CancellationToken cancellationToken)
             {
-                // Process all metadata references in parallel.
-                var tasks = project.MetadataReferences.OfType<PortableExecutableReference>()
-                                   .Select(r => UpdateReferenceAsync(project, r, cancellationToken))
-                                   .ToArray();
+                // Process all metadata references. If it remote workspace, do this in parallel.
+                var tasks = new List<Task>();
+                var isRemoteWorkspace = IsRemoteWorkspace(project);
+
+                foreach (var reference in project.MetadataReferences.OfType<PortableExecutableReference>())
```

Comment:
you can use http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/StateChecksums.cs,108

instead of you re-calculating the checksum if you want to. order of checksum will be same as Project.MetadataReferences's order.